api/bases/glance.openstack.org_glances.yaml

```diff
@@ -5950,6 +5966,14 @@ spec:
                       to register in glance
                     type: string
                 type: object
+              networkAttachmentDefinitions:
+                description: NetworkAttachmentDefinitions list of network attachment
+                  definitions the service pod of extenal and internal API instances
+                  get attached to. Define them on the instance level will have no
+                  affect to overwrite the global ones.
```

Comment:
or would it make sense that the 2 api instances could have different network attachments? right now I don't think so.

Reply:
Could each instance receive a different list of NADs? So the patch only adds one NAD list parameter but the list itself differs for internal vs external glance instances?

External:
```
k8s.v1.cni.cncf.io/networks: '[{"name": "publicapi", "namespace": "openstack"},{"name": "storage", "namespace": "openstack"}]'
```
Internal:
```
k8s.v1.cni.cncf.io/networks: '[{"name": "internalapi", "namespace": "openstack"},{"name": "storage", "namespace": "openstack"}]'
```

Reply:
I don't know how this will all shake out in the end, but it definitely feels right that the internal and external glance services will need access to different networks.

Reply:
I basically agree with the fact that we should have two separate `NADs` for internal/external: eventually, if at some point we decide to just run a single instance (e.g. when `file` is used as backend), we can merge the two structures but it would be cleaner having `NetworkAttachmentDefinitionInternal` and `NetworkAttachmentDefinitionExternal` in the top-level CR and assign the right one to the underlying service when the `Deployment` is created.

@stuggi not sure about your thoughts here: as long as the networking picture is not well defined I guess it's ok having this patch merged and follow up on the internal/external split.
With this change the two `GlanceAPI` instances will receive all the networks defined, but given you're doing the change it would be great setting up split mode for the network as well.

Reply:
For the initial version I guess we could then just have the two in the service spec, remove it from the top level glance and the db-init job from glance controller use the one from the internal glanceapi spec for its job?

Reply:
That sounds good.